lib/util.c

```diff
@@ -1460,6 +1460,64 @@ grn_inspect_limited(grn_ctx *ctx, grn_obj *buffer, grn_obj *obj)
   return buffer;
 }
 
+grn_obj *
+grn_inspect_key(grn_ctx *ctx, grn_obj *buffer, grn_obj *table, const void *key, unsigned int key_size)
+{
+  if (!buffer) {
+    buffer = grn_obj_open(ctx, GRN_BULK, 0, GRN_DB_TEXT);
+  }
+
+  if (!table) {
+    GRN_TEXT_PUTS(ctx, buffer, "(NULL)");
+    return buffer;
+  }
+
+  if (table->header.type == GRN_TABLE_NO_KEY) {
+    grn_obj table_type, table_name;
+    GRN_TEXT_INIT(&table_type, 0);
+    GRN_TEXT_INIT(&table_name, 0);
+    grn_table_type_inspect(ctx, &table_type, table);
+    grn_inspect_name(ctx, &table_name, table);
+    grn_text_printf(ctx, buffer, "#<key: (nil) table:#<%.*s %.*s>>",
+                    (int)GRN_TEXT_LEN(&table_type), GRN_TEXT_VALUE(&table_type),
+                    (int)GRN_TEXT_LEN(&table_name), GRN_TEXT_VALUE(&table_name));
+    GRN_OBJ_FIN(ctx, &table_type);
+    GRN_OBJ_FIN(ctx, &table_name);
+    return buffer;
```

Comment:
We don't need to create temporary buffer:

```c
GRN_TEXT_PUTS(ctx, buffer, "#<key: (nil) table:#<");
grn_table_type_inspect(ctx, buffer);
GRN_TEXT_PUTS(ctx, buffer, " ");
grn_inspect_name(ctx, buffer, table);
GRN_TEXT_PUTS(ctx, buffer, ">>");
```

Reply:
I got it. I've fixed.